File: array/849.py
class Solution:
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        if not seats:
            return 0
        pre = 0
        nxt = 0
        ret = 0
        cur = 0
        
        for i in range(0, len(seats)):
            if seats[i] == 0:
                pre = i-1
                nxt = i+1
                if i == 0:
                    while nxt < len(seats) and seats[nxt] == 0:
                        nxt += 1
                    cur = nxt - i
                    ret = max(cur, ret)
                elif i == len(seats) - 1:
                    while seats[pre] == 0:
                        pre -= 1
                    cur = i - pre
                    ret = max(cur, ret)
                else:
                    while seats[pre] == 0:
                        pre -= 1

                    while nxt < len(seats) and seats[nxt] == 0:
                        nxt += 1

                    print(i-pre)
                    cur = min((i-pre), (nxt-i))
                    # print(cur)
                    ret = max(cur, ret)
        return ret
        

File: array/849.cc
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int begin = -1, end = 0, ret = 0, cur = 0;
        for (int i = 0; i < seats.size(); ++i) {
            if (seats[i] == 1) {
                begin = i;
            } else {
                end = begin+1;
                while (end < seats.size() && seats[end] == 0)
                    end++;
                if (begin == -1 || end == seats.size()) {
                    ret = max(ret, end-begin-1);
                } else {
                    if ((end+1-begin) % 2 == 0) {
                        ret = max(ret, (end+1-begin)/2-1); 
                    } else {
                        ret = max(ret, (end+1-begin)/2);     
                    }   
                }
                begin = end;
                i = end;
                
            }
        }
        return ret;
    }
};


class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int last1 = -1, ret = 0;
        int len = seats.size();
        for (int i = 0; i < len; ++i) {
            if (seats[i] == 1) last1 = i;
            if (last1 == -1) {
                ret = i+1;
            } else {
                ret = std::max(ret, (i-last1+1)/2);
            }
        }
        ret = std::max(ret, len-last1-1);
        return ret;
    }
};
